pyccel/ast/literals.py

```diff
@@ -250,9 +251,9 @@ def _collect_python_val(arg):
             The Python value of the argument.
         """
         if isinstance(arg, Literal):
-            return arg.python_value
-        elif isinstance(arg, (int, float)):
-            return arg
+            return float(arg.python_value)
+        elif isinstance(arg, (int, float, np.integer, np.floating)):
+            return float(arg)
```

Comment:
I had not realized that this code is inside class `LiteralComplex`...

Reply:
`LiteralComplex` takes 2 floats, one for the real part and one for the imag part. This function unpacks them

Reply:
I would not say "unpack", but I understand now. It is separately called on the real and imaginary parts of the complex number. I find the docstring of this method a bit confusing as it only reads "input argument" but it does not mention real and imaginary parts.

Reply:
6fbc375a6